### ISSUE TITLE
.NET support

### ISSUE THREAD

**Issue by brancz** [parca-dev/parca-agent#161]
.NET (dotnet) support has been requested a number of times now.

It should already work with perf-map enabled: https://docs.microsoft.com/en-us/dotnet/core/run-time-config/debugging-profiling#write-perf-map

That said, it should also work automatically without any change to a user's deployment.

**Comment by LarsBingBong** [comment]
Any news on this?

**Comment by brancz** [comment]
Perhaps this is what you meant, but just in case if not, we did recently add an example of what to do to get .NET application working today: https://github.com/parca-dev/parca-demo/pull/18

We're still trying to figure out why ahead-of-time compiled .NET binaries aren't working (they should to our knowledge, it might be something small): https://github.com/parca-dev/parca-demo/pull/23

The ultimate goal is that what needs to be enabled in #18 is not necessary, which we haven't started the work for yet, so in the meantime that is the best way to use Parca with .NET.

**Comment by LarsBingBong** [comment]
ahead-of-time compiled ... that's good old CI stage build *.dll files containing dotnet code? Is that what you mean? Thanks.

**Comment by brancz** [comment]
I know next to nothing about .NET, so I can only go by docs or other web searches. From the docs, I understand that ahead-of-time compilation to a regular self-contained Linux ELF binary (which should work just fine with Parca Agent if debuginfos are available) is only [supported since .NET 7](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/), which appears to have only been [released on November 8th, 2022](https://dotnet.microsoft.com/en-us/download/dotnet/7.0), so it appears to me it's rather the bleeding edge than an old feature.

**Comment by LarsBingBong** [comment]
Thank you for taking the time @brancz to look into this even further. Much appreciated. I'll see what I can get going here on the tracing side of things when workloads are coded in dotnet. Thank you!

**Comment by maxbrunet** [comment]
Yes, AOT is the bleeding edge, you are likely interest by what's here (JIT): https://github.com/parca-dev/parca-demo/tree/main/dotnet (See link in README and `ENV` in Dockerfile)
